test/WebSites/ValidationWebSite/ProductValidatorAttibute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+using ValidationWebSite.ViewModels;
+
+namespace ValidationWebSite
+{
+    public class ProductValidatorAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+
+            if (value is ProductViewModel)
+            {
+                var product = (ProductViewModel)value;
```

Comment:
nit: demonstrate correct use of conditional and unconditional casts:
``` c#
  var product = value as ProductViewModel;
  if (product != null)
  {
    ...
  }
  else
  {
    var software = (SoftwareViewModel)value;
    ..
  }
```

could be extra-careful and use `as` for SoftwareViewModel as well. then return a `ValidationResult` explaining the validator didn't get an instance of the expected type.  right now, that case will throw an `InvalidCastException`.